mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/model/dependency/PojoTypeDependencyContext.java

```diff
@@ -0,0 +1,94 @@
+	 * or that do not rely on the bridged element at all (constant bridges, bridges adding the last indexing date, ...).
+	 * <p>
+	 * Note that calling this method prevents from declaring any other dependency,
+	 * and trying to do so will trigger an exception.
+	 */
+	void useRootOnly();
+
+	/**
+	 * Declare that the given path is read by the bridge at index time to populate the indexed document.
+	 *
+	 * @param pathFromBridgedTypeToUsedValue The path from the bridged type to the value used by the bridge,
+	 * as a String. The string is interpreted with default value extractors: see {@link PojoModelPath#parse(String)}.
+	 * @return {@code this}, for method chaining.
+	 * @throws org.hibernate.search.util.common.SearchException If the given path cannot be applied to the bridged type.
+	 * @see #use(PojoModelPathValueNode)
+	 */
+	default PojoTypeDependencyContext use(String pathFromBridgedTypeToUsedValue) {
+		return use( PojoModelPath.parse( pathFromBridgedTypeToUsedValue ) );
+	}
+
+	/**
+	 * Declare that the given path is read by the bridge at index time to populate the indexed document.
+	 * <p>
+	 * Every component of this path will be considered as a dependency,
+	 * so it is not necessary to call this method for every subpath.
+	 * In other words, if the path {@code "myProperty.someOtherPropety"} is declared as used,
+	 * Hibernate Search will automatically assume that {@code "myProperty"} is also used.
+	 *
+	 * @param pathFromBridgedTypeToUsedValue The path from the bridged type to the value used by the bridge.
+	 * @return {@code this}, for method chaining.
+	 * @throws org.hibernate.search.util.common.SearchException If the given path cannot be applied to the bridged type.
+	 */
+	PojoTypeDependencyContext use(PojoModelPathValueNode pathFromBridgedTypeToUsedValue);
+
+	/**
+	 * Start the declaration of dependencies to properties of another entity,
+	 * without specifying the path to that other entity.
+	 * <p>
+	 * <strong>Note:</strong> this is only useful when the path from the bridged type to that other entity
+	 * cannot be easily represented, but the inverse path can.
+	 * For almost all use cases, this method won't be useful and calling {@link #use(String)} will be enough.
+	 *
+	 * @param otherEntityType The raw type of the other entity.
+	 * @param pathFromOtherEntityTypeToBridgedType The path from the other entity type to the bridged type,
+	 * as a String. The string is interpreted with default value extractors: see {@link PojoModelPath#parse(String)}.
+	 * Used when the other entity changes, to collect the instances that must be reindexed.
+	 * @return A context allowing to declare which properties
+	 * @throws org.hibernate.search.util.common.SearchException If the bridged type is not an entity type,
+	 * or the given type is not an entity type,
+	 * or the given path cannot be applied to the given entity type.
+	 */
+	default PojoOtherEntityDependencyContext fromOtherEntity(Class<?> otherEntityType,
```

Comment:
You're right that the name is not very good, but I can't find a better one for now. "inverseAssociation" almost gets there, but not quite, since we can hardly call it an "inverse association" when there is no "association" to be inversed.

I'll mark all these APIs as experimental in #1963 , so we can revisit the names later.